### PR TITLE
logout() only if connected #3123

### PR DESCRIPTION
**Pull request by urajio**
https://github.com/spring-projects/spring-integration/issues/3123
#3123 send logout to ftp server only if connected

### CODE REVIEW

**Comment by pivotal-issuemaster** [comment]
@urajio Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=3126)!

[Click here](https://cla.pivotal.io/sync/spring?repositoryId=spring-projects/spring-integration&pullRequestId=3126) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

**Comment by pivotal-issuemaster** [comment]
@urajio Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/spring?repositoryId=spring-projects/spring-integration&pullRequestId=3126)!

File: spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java [hunk 1]
```diff
@@ -154,7 +154,8 @@ public void close() {
 			if (this.readingRaw.get() && !finalizeRaw() && LOGGER.isWarnEnabled()) {
 				LOGGER.warn("Finalize on readRaw() returned false for " + this);
 			}
-			this.client.logout();
+			if (this.client.isConnected())
```
Comment (artembilan) [inline spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java]: Checkstyle violation:
```
[ant:checkstyle] [ERROR] /home/travis/build/spring-projects/spring-integration/spring-integration-ftp/src/main/java/org/springframework/integration/ftp/session/FtpSession.java:157: 'if' construct must use '{}'s. [NeedBraces]
```

You can verify yourself locally using `gradlew clean :spring-integration-ftp:check`

**Review by artembilan — CHANGES_REQUESTED** [review]
Wanted to ask about a test-case on the matter, but doesn't look it worth that.

Also, please, add your name to `@author` list in that modified `FtpSession` class.

**Comment by artembilan** [comment]
Looks like your change has affected some test:
```
org.springframework.integration.ftp.session.SessionFactoryTests > testFtpClientInteraction() FAILED
    org.mockito.exceptions.verification.WantedButNotInvoked at SessionFactoryTests.java:79
```
Would you mind to take a look and adjust respectively?
I guess that test just doesn't have a session as connected so we don't call `logout()` according a new logic you have introduced.
I don't say that your change is wrong, but I believe the test must be fixed therefore.

Thanks

**Comment by urajio** [comment]
sorry, i'm not sure how to fix this properly

**Comment by artembilan** [comment]
> sorry, i'm not sure how to fix this properly

No, problem. I'll take a look today and merge your PR with an appropriate fix for the test.

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/c6358a93d44f0e132f0e94d62379983adef5e4ce

@urajio ,
thank you for contribution; looking forward for more!

**Pull request closed without merging**